src/flowcean/environments/directory_loader.py

```diff
@@ -0,0 +1,44 @@
+from collections.abc import Callable, Iterable
+from pathlib import Path
+
+from flowcean.core.environment.offline import OfflineEnvironment
+
+LoaderFunction = Callable[[Path], OfflineEnvironment]
+
+
+def load_directory(
```

Comment:
I'm not sure whether this function does what it says. At least for now it does not load anything (and that's a good thing). In the end it just maps a function to a list of paths... I feel like we might not even need that function in the end.

The only functionality it provides is filtering for files or directories. But this is also something I doubt is designed wisely: What about symbolic links for example?

Without such a function the user would just write:
```python
environments = map(CsvEnvironment, Path("my_data").glob("*.csv"))
```
This is something we are able to do in `main` and at least to my eyes this is already simple enough.

TLDR, I'd suggest to not provide such a function because it is not actually doing anything...
What do you think?